# HG changeset patch
# User Matti Picus <matti.picus@gmail.com>
# Date 1691871129 -10800
#      Sat Aug 12 23:12:09 2023 +0300
# Branch conda-3.9-v7.3.12
# Node ID 80484c5f1f9cea6785223928bfe33e324cd3d166
# Parent  744a279accbe082338dd6aba52332b60a3d56863
fix edge case of datetime isoformat parsing (issue 3989)

diff -r 744a279accbe -r 80484c5f1f9c lib-python/3/datetime.py
--- a/lib-python/3/datetime.py	Fri Aug 11 11:34:49 2023 +0300
+++ b/lib-python/3/datetime.py	Sat Aug 12 23:12:09 2023 +0300
@@ -270,6 +270,8 @@
 def _parse_isoformat_date(dtstr):
     # It is assumed that this function will only be called with a
     # string of length exactly 10, and (though this is not used) ASCII-only
+    if len(dtstr) < 10:
+        raise ValueError('isoformat expects a string of length 10')
     year = int(dtstr[0:4])
     if dtstr[4] != '-':
         raise ValueError('Invalid date separator: %s' % dtstr[4])
